### ISSUE TITLE
Atlas doesn't support creating concepts sets using source_concept_ids

### ISSUE THREAD

**Issue by aostropolets** [OHDSI/Atlas#950]
There is no possibility to restrict a concept set to the particular concepts using source_concept_id.
The issue is in two parts:
1. If a concept doesn't have a standard counterpart in current vocabularies, the CDM tables *_concept_ids are populated with 0. While we preserve source data in source_concept_ids, we can't use them to create a concept set.
2. If a concept is mapped to a generic one, we want to restrict our search to specific source concept_ids.

### Actual behavior
No search across source_concept_ids is embedded.
![default](https://user-images.githubusercontent.com/17028852/46269137-11485580-c50d-11e8-995b-f2253d8857e7.png)
![default](https://user-images.githubusercontent.com/17028852/46269176-684e2a80-c50d-11e8-9aba-838ad9e60c15.png)
![default](https://user-images.githubusercontent.com/17028852/46269205-ad725c80-c50d-11e8-804f-80b7bde867f7.png)

**Comment by chrisknoll** [comment]
Leave the main concept 'any concept' (so that we won't look for any concept_id in the 'standard' concept column, and add a 'source_concept' criterion and specify the concept set with source concepts in it.  This will look for the concept in the source_concept_id field.


Would that do as you wanted?

**Comment by chrisknoll** [comment]
Hi, @aostropolets , 
  I'm not sure I get your example using an ATC class:  you won't be able to find non-standard concepts looking at concept ancestor (via descendants).  You can always have a concept set expression which is filled with a set of non-standard concepts and also the ATC concept + descendants: that will resolve into the set of concepts with your specified non-standard concepts + the concepts from the descendants of the ATC concept.

Currently, if you specify both the standard conceptset AND the source concept set filter, the logic of the query is to do an OR between them ie:
```
JOIN #codesets codesets on ((de.drug_concept_id in (select concept_id from #codesets where codeset_id = 1) OR (de.drug_source_concept_id in (select  concept_id from #codesets where codeset_id = 2))
```
but, as @pbr6cornell  has pointed out: the OR in the join clause is slower, but also not what you'd normally expect:  it should say that both the drug_concept_id AND drug_source_concept_id should be found in their respective concept sets for the row to be returned. If you want 'OR', you'd define your cohort definition to select BOTH the drug exposures from the one concept and the drug exposures with the source_concept_id from the other.

So, we will push this fix out in the next release, but you should do the following to get the result you want:

In english: select records that have drug_concept_id = 0 and drug_source_concept_id in (set of concept IDs):

ConceptSet: Unknown
- Contains 1 Concept Id: 0

ConceptSet: Non Standard ATC Drugs
- Contains the set of ATC drug source concepts you want to lookup.

Your criteria will be defined as:
Select drug_exposures having 
- drug exposure of 'Unknown'
- and drug_source_concept of 'Non Standard ATC Drugs'

After we make the logical change that @pbr6cornell requested: that will do an AND between both concept set checks, not an OR.

**Comment by aostropolets** [comment]
Ok, not to confuse you with classification concepts, we can take any other non-standard code, for instance, 43251842 Xofigo. From what you're saying, I can conclude that with we'll be able to find it in the device_source_concept_id starting from the next release. I do agree that AND query will be more beneficial not only from the computational perspective but also due to the fact that some of the vendors still put non-standard IDs in their *_concept_id.

**Comment by chrisknoll** [comment]
This is supported as of Circe 1.7.0, which has been included in WebAPI 2.6.0.
